PWGHF/TableProducer/candidateSelectorXicToPKPi.cxx

```diff
@@ -239,15 +290,45 @@ struct HfCandidateSelectorXicToPKPi {
 
       if (pidXicToPKPi == 0 && pidXicToPiKP == 0) {
         hfSelXicToPKPiCandidate(statusXicToPKPi, statusXicToPiKP);
+        if (applyMl) {
+          hfMlXicToPKPiCandidate(outputMlXicToPKPi, outputMlXicToPiKP);
+        }
         continue;
       }
-
       if ((pidXicToPKPi == -1 || pidXicToPKPi == 1) && topolXicToPKPi) {
         statusXicToPKPi = 1; // identified as Xic->pKpi
       }
       if ((pidXicToPiKP == -1 || pidXicToPiKP == 1) && topolXicToPiKP) {
         statusXicToPiKP = 1; // identified as Xic->piKp
       }
+      SETBIT(statusXicToPKPi, aod::SelectionStep::RecoPID);
+      SETBIT(statusXicToPiKP, aod::SelectionStep::RecoPID);
+
+      if (applyMl) {
+        // ML selections
+        bool isSelectedMlXicToPKPi = false;
+        bool isSelectedMlXicToPiKP = false;
+
+        if (statusXicToPKPi > 0) {
```

Comment:
This is not enough to guarantee that both topological and PID selections are satisfied, given the current code.